chart/skywalking/templates/oap-cm-override-application.yaml

```diff
@@ -0,0 +1,27 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{- if (.Files.Glob "files/conf.d/oap/application.yml") }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "skywalking.fullname" . }}-override-application-yml
+  labels:
+    app: {{ template "skywalking.name" . }}
+    release: {{ .Release.Name }}
+    component: {{ .Values.oap.name }}
+data:
+{{ (.Files.Glob "files/conf.d/oap/application.yml").AsConfig | indent 2 }}
```

Comment:
We had better support other configuration files more than `applicaiton.yml`. 

Overriding `application.yml` is a legacy operation and doesn't recommend cloud-native users to opt to. That's because the main configuration file(`application.yml`) is the combination of most core components and modules varies in different versions. so we should introduce others, for instance, alarm rule, log4j and etc firstly. 

From my experience of operating oap, the log4j XML configuration file is at the top of debugging tool list. Based on that, I prefer to include it at leaset

Reply:
> Overriding application.yml is a legacy operation and doesn't recommend cloud-native users to opt to. That's because the main configuration file(application.yml) is the combination of most core components and modules varies in different versions

I could understand your idea, but can't agree. If you take a look at `Configuration Vocabulary` doc, there are many many configurations not included in the released `application.yml`. We don't have the chance to verify them every time. So, overriding the core YAML is super important.

Reply:
> From my experience of operating oap, the log4j XML configuration file is at the top of debugging tool list. Based on that, I prefer to include it at leaset
 
log4j2.xml and some other files that I think are necessary have been added, and because `application.yml` is the first requirement that our users (from QQ group) requested, I'll just left it there